libs/SmartStore/SmartStore/Classes/SFSmartStore.m

```diff
@@ -205,11 +209,16 @@ - (id) initWithName:(NSString*)name user:(SFUserAccount *)user isGlobal:(BOOL)is
                                                 this->_dataProtectionKnownAvailable = NO;
                                             }];
         
-        _soupNameToTableName = [[NSMutableDictionary alloc] init];
-        _attrSpecBySoup = [[NSMutableDictionary alloc] init];
-        _indexSpecsBySoup = [[NSMutableDictionary alloc] init];
+        _soupNameToTableName = [[NSCache alloc] init];
+        _soupNameToTableName.countLimit = CACHES_COUNT_LIMIT;
+        
+        _attrSpecBySoup = [[NSCache alloc] init];
+        _attrSpecBySoup.countLimit = CACHES_COUNT_LIMIT;
         
-        _smartSqlToSql = [[NSMutableDictionary alloc] init];
+        _indexSpecsBySoup = [[NSCache alloc] init];
+        _indexSpecsBySoup.countLimit = CACHES_COUNT_LIMIT;
+        
+        _smartSqlToSql = [[SFSmartSqlCache alloc] initWithCountLimit:CACHES_COUNT_LIMIT];
```

Comment:
@wmathurin shouldn't the number of objects  held in cache be SFSmartSqlCache  be higher as compared to the other caches?

Reply:
@wmathurin Should we provide a way to clear the caches? What would happen if the soup features changed?

Reply:
I don't want to make clearing the caches a public api.
I need to check the code to see if altering a soup causes the cache to be cleared or not (it might not).

Reply:
https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/2909/commits/2d55ca969ded6e5948c14ac9cfa585134c543f45 takes care of clearing caches when soup is changed.